### PR TITLE
8341015: OopStorage location decoder crashes accessing non-initalized OopStorage

### PR DESCRIPTION
**Pull request by shipilev**
When debugging CDS, I asked for `os::print_location` before OopStorage was initialized, like an error handler would do. This is a fairly unusual situation, this is why we have not seen it before. Anyhow, we entered the new code added by [JDK-8340392](https://bugs.openjdk.org/browse/JDK-8340392), which crashed on `OopStorage` that was `nullptr`. I think we should null-check `OopStorage` before calling into it.

Additional testing:
 - [x] OopStorageSetTest still passing
 - [x] Verified the check is now passing in similar debugging session

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341015](https://bugs.openjdk.org/browse/JDK-8341015): OopStorage location decoder crashes accessing non-initalized OopStorage (**Enhancement** - P4)

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21204/head:pull/21204` \
`$ git checkout pull/21204`

Update a local copy of the PR: \
`$ git checkout pull/21204` \
`$ git pull https://git.openjdk.org/jdk.git pull/21204/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21204`

View PR using the GUI difftool: \
`$ git pr show -t 21204`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21204.diff">https://git.openjdk.org/jdk/pull/21204.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21204#issuecomment-2376707062)

### CODE REVIEW

File: src/hotspot/share/gc/shared/oopStorageSet.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ bool OopStorageSet::print_containing(const void* addr, outputStream* st) {
   if (addr != nullptr) {
     const void* aligned_addr = align_down(addr, alignof(oop));
     for (OopStorage* storage : Range<Id>()) {
-      if (storage->print_containing((oop*) aligned_addr, st)) {
+      if (storage != nullptr && storage->print_containing((oop*) aligned_addr, st)) {
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorageSet.cpp:89]: Add a comment?  Something like "Might get here while handling error before storage initialization."

**Review by kimbarrett — APPROVED** [review]
Looks good.

File: src/hotspot/share/gc/shared/oopStorageSet.cpp (hunk 1 above)
Comment (shipilev) [reply src/hotspot/share/gc/shared/oopStorageSet.cpp]: Sure thing, see new commit.

Comment (tschatzl) [reply src/hotspot/share/gc/shared/oopStorageSet.cpp]: Another maybe preferable option could be printing "uninitialized" or something.

**Review by tschatzl — APPROVED** [review]
lgtm, maybe it's worth to explicitly print an "unitialized" message

File: src/hotspot/share/gc/shared/oopStorageSet.cpp (hunk 1 above)
Comment (kimbarrett) [reply src/hotspot/share/gc/shared/oopStorageSet.cpp]: @tschatzl I don't think printing "uninitialized" or anything like that is really appropriate here.  What this code
is doing is printing something if-and-only-if the pointer is found to be in an oopstorage block.  There aren't
any of those if there's no oopstorage yet.

**Review by kimbarrett — APPROVED** [review]
Still looks good.

**Comment by shipilev** [comment]
> lgtm, maybe it's worth to explicitly print an "unitialized" message

A normal thing to do in these printers is to silently return, letting other printers to handle the location. If OopStorage does not recognize the pointer, the downstream NMT and generic SafeFetch code would try to look it up.

**Comment by shipilev** [comment]
Thanks for reviews, I think this is simple enough to push on Friday.

/integrate

**Pull request closed without merging**